templates/components/engines/typescript/agent/tools/index.ts

```diff
@@ -0,0 +1,20 @@
+import { BaseToolWithCall } from "llamaindex";
+import { tools as weatherTools } from "./weather";
+
+const functionTools: Record<string, BaseToolWithCall[]> = {
+  weather: weatherTools,
+};
+
+export function getFunctionTools(
+  localConfig: Record<string, unknown>,
+): BaseToolWithCall[] {
+  const tools: BaseToolWithCall[] = [];
+
+  Object.keys(localConfig).forEach((key) => {
+    if (key in functionTools) {
+      tools.push(...functionTools[key]);
```

Comment:
@thucpn although i like the `Functiontool.from` idea, in general, we will need to create a tool with the `localConfig[key]` passed to the constructor (otherwise our config object also doesn't make a lot of sense).

so better turn the weather tool into a class implementing `BaseTool`

Reply:
Or support BaseTool tools and Function tools in the factory like the Python version